mutiny-core/src/lib.rs

```diff
@@ -653,33 +660,9 @@ impl<S: MutinyStorage> MutinyWallet<S> {
         let invoice = if self.config.safe_mode {
             None
         } else {
-            // Check if a federation exists
-            let federation_ids = self.list_federation_ids().await?;
-            if !federation_ids.is_empty() {
-                // Use the first federation for simplicity
-                let federation_id = &federation_ids[0];
-                let fedimint_client = self.federations.lock().await.get(federation_id).cloned();
-
-                match fedimint_client {
-                    Some(client) => {
-                        // Try to create an invoice using the federation
-                        match client
-                            .get_invoice(amount.unwrap_or_default(), labels.clone())
-                            .await
-                        {
-                            Ok(inv) => Some(inv.bolt11.ok_or(MutinyError::WalletOperationFailed)?),
-                            Err(_) => None, // Handle the error or fallback to node_manager invoice creation
-                        }
-                    }
-                    None => None, // No federation client found, fallback to node_manager invoice creation
-                }
-            } else {
-                // Fallback to node_manager invoice creation if no federation is found
-                let inv = self
-                    .node_manager
-                    .create_invoice(amount, labels.clone())
-                    .await?;
-                Some(inv.bolt11.ok_or(MutinyError::WalletOperationFailed)?)
+            match self.create_lightning_invoice(amount, labels.clone()).await {
+                Ok(invoice) => Some(invoice), // Assuming create_lightning_invoice returns a Result
+                Err(_) => None, // You can handle the error more specifically if needed
```

Comment:
We were returning an error here before, should probably keep that functionality at least for now.
otherwise this is the exact same as `ok()`

Reply:
We didn't return error here, when we create bip21, we return the address that's created but without the invoice, and display a warning.

Reply:
This is what it was before this PR: 

```
        } else {
            match self.create_lightning_invoice(amount, labels.clone()).await {
                Ok(invoice) => Some(invoice),
                Err(_) => None,
            }
        };
```

I was incorrect in making it return an error when I introduced fedimint.

Reply:
Okay, we can at least clean this up a bit with `ok()` if you want

Reply:
Doing `ok().and_then()` here now.